src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java

```diff
@@ -589,28 +690,62 @@ public CauseManagement getAncestorCauseManagement() {
 
     @Override
     @JsonIgnore
+    @DynamoDBIgnore
     public String getIconFileName() {
         return PluginImpl.getDefaultIcon();
     }
 
     @Override
     @JsonIgnore
+    @DynamoDBIgnore
     public String getDisplayName() {
         return name;
     }
 
     @Override
     @JsonIgnore
+    @DynamoDBIgnore
     public String getUrlName() {
         return id;
     }
 
 
     @Override
+    @DynamoDBIgnore
     public FailureCauseDescriptor getDescriptor() {
         return Jenkins.getInstance().getDescriptorByType(FailureCauseDescriptor.class);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) { return true; }
+        if (!(o instanceof FailureCause)) {
+            return false;
+        }
+
+        FailureCause fc = (FailureCause)o;
+        Field[] fields = this.getClass().getDeclaredFields();
+        EqualsBuilder eb = new EqualsBuilder();
+        for (Field f:fields) {
+            try {
+                eb.append(f.get(this), f.get(fc));
+            } catch (IllegalArgumentException | IllegalAccessException e) {
+                return false;
+            }
+        }
+        return eb.isEquals();
+    }
+
+    /**
+     * Makes checkstyle happy.
+     * @return hashcode of class
+     */
+    @Override
+    public int hashCode() {
+        //Making checkstyle happy.
+        return getClass().getName().hashCode();
```

Comment:
Same equals and hashCode problems here, but in this case just looking at the `id` might be enough.